packages/jest-runtime/src/index.ts

```diff
@@ -314,7 +315,11 @@ export default class Runtime {
         ? new RegExp(ignorePatternParts.join('|'))
         : undefined;
 
-    return new HasteMap({
+    const HasteMapClass = config.haste.hasteMapModulePath
+      ? (require(config.haste.hasteMapModulePath) as HasteMapStatic)
```

Comment:
using `require` means it cannot use ESM, and it being inline means we cannot use `import` without changing the signature of the function. Could you refactor this so we load `hasteMapModulePath` some other place where it can be async?

this applies to the other `require`s added as well